imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -171,14 +204,46 @@ def calculate_calibration_factor(time: int) -> None:
 
     Parameters
     ----------
-    time : int
-        Input time.
-    """
-    # NOTE: waiting on fake calibration data to write this.
-    pass
-
+    acquisition_time : float
+        Acquisition time in S/C MET.
 
-def apply_in_flight_calibration(data: np.ndarray) -> None:
+    Returns
+    -------
+    cal_factor : numpy.ndarray
+        Calibration factor for each 7 CEM detectors. Array shape is (7,).
+    """
+    # Read in in-flight calibration data
+    in_flight_cal_df = read_in_flight_cal_data()
+    # For the given acquisition time, find it's nearest pre and post time
+    pre_cal_time = (
+        in_flight_cal_df[in_flight_cal_df["met_time"] < acquisition_time]
+        .iloc[-1]
+        .values[0]
+    )
+    post_cal_time = (
+        in_flight_cal_df[in_flight_cal_df["met_time"] > acquisition_time]
+        .iloc[0]
+        .values[0]
+    )
+    run = post_cal_time - pre_cal_time
```

Comment:
```suggestion
    cal_time_length = post_cal_time - pre_cal_time
```

This variable name was a bit confusing to me so just suggesting something different here